File: CS-Examples/11_Formatting/HighlightRankedValues.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet
Imports Spire.Xls.Core.Spreadsheet.Collections
Imports Spire.Xls.Core

Namespace HighlightRankedValues
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an existing Excel file into the workbook
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_6.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Create a new collection of conditional formats for the worksheet
            Dim xcfs As XlsConditionalFormats = sheet.ConditionalFormats.Add()

            ' Define the range of cells to apply conditional formatting to
            xcfs.AddRange(sheet.Range("D2:D10"))

            ' Add a top/bottom conditional format with "Top" type and show top 2 values
            Dim format1 As IConditionalFormat = xcfs.AddTopBottomCondition(TopBottomType.Top, 2)

            ' Set the format type to Top/Bottom
            format1.FormatType = ConditionalFormatType.TopBottom

            ' Set the background color for the conditional format
            format1.BackColor = Color.Red

            ' Create another collection of conditional formats for the worksheet
            Dim xcfs1 As XlsConditionalFormats = sheet.ConditionalFormats.Add()

            ' Define the range of cells to apply conditional formatting to
            xcfs1.AddRange(sheet.Range("E2:E10"))

            ' Add a top/bottom conditional format with "Bottom" type and show bottom 2 values
            Dim format2 As IConditionalFormat = xcfs1.AddTopBottomCondition(TopBottomType.Bottom, 2)

            ' Set the format type to Top/Bottom
            format2.FormatType = ConditionalFormatType.TopBottom

            ' Set the background color for the conditional format
            format2.BackColor = Color.ForestGreen

            ' Specify the filename to save the modified workbook
            Dim result As String = "Result-HighlightTopAndBottomRankedValues.xlsx"

            ' Save the workbook to the specified file path using Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/11_Formatting/HighlightRankedValues.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;
using Spire.Xls.Core.Spreadsheet.Collections;
using Spire.Xls.Core;

namespace HighlightRankedValues
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a workbook.
			Workbook workbook = new Workbook();

            //Load the file from disk.
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_6.xlsx");

            //Get the first worksheet.
			Worksheet sheet = workbook.Worksheets[0];

            //Apply conditional formatting to range ¡°D2:D10¡± to highlight the top 2 values.
            XlsConditionalFormats xcfs = sheet.ConditionalFormats.Add();
            xcfs.AddRange(sheet.Range["D2:D10"]);
            IConditionalFormat format1 = xcfs.AddTopBottomCondition(TopBottomType.Top, 2);
            format1.FormatType = ConditionalFormatType.TopBottom;
            format1.BackColor = Color.Red;

            //Apply conditional formatting to range ¡°E2:E10¡± to highlight the bottom 2 values.
            XlsConditionalFormats xcfs1 = sheet.ConditionalFormats.Add();
            xcfs1.AddRange(sheet.Range["E2:E10"]);
            IConditionalFormat format2 = xcfs1.AddTopBottomCondition(TopBottomType.Bottom,2);
            format2.FormatType = ConditionalFormatType.TopBottom;
            format2.BackColor = Color.ForestGreen;

            // Specify the output file name.
            String result = "Result-HighlightTopAndBottomRankedValues.xlsx";

            //Save to file.
            workbook.SaveToFile(result, ExcelVersion.Version2013);
            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}
